src/hardware/mouse/mouse_config.cpp

```diff
@@ -0,0 +1,350 @@
+/*
+ *  Copyright (C) 2022-2022  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include "mouse_config.h"
+#include "mouse_common.h"
+#include "mouse_interfaces.h"
+
+#include "checks.h"
+#include "control.h"
+#include "math_utils.h"
+#include "setup.h"
+#include "string_utils.h"
+#include "support.h"
+#include "video.h"
+
+#include <cmath>
+
+CHECK_NARROWING();
+
+// TODO - IntelliMouse Explorer emulation is currently deactivated - there is
+// probably no way to test it. The IntelliMouse 3.0 software can use it, but
+// it seems to require physical PS/2 mouse registers to work correctly,
+// and these are not emulated yet.
+
+// #define ENABLE_EXPLORER_MOUSE
+
+MouseConfig mouse_config;
+MousePredefined mouse_predefined;
+
+static const std::vector<std::string> list_models_ps2 = {
+        "standard",
+        "intellimouse",
+#ifdef ENABLE_EXPLORER_MOUSE
+        "explorer",
+#endif
+};
+
+static const std::vector<std::string> list_models_com = {
+        "2button",
+        "3button",
+        "wheel",
+        "msm",
+        "2button+msm",
+        "3button+msm",
+        "wheel+msm",
+};
```

Comment:
Suggest lifting these strings to their own constexprs:

```

auto model_2_button_name = "2button";

auto model_3_button_name = "3button";

auto model_wheel_name = "wheel";

...

```

Then these can be used to construct the vector's content and also do direct string-to-char[] comparisons in the all the if tests below (instead of using harcoded indexes).

Reply:
Did something like that, also for the PS/2 mouse models. Additionally, de-duplicated some code.